### ISSUE TITLE
Error x509: certificate signed by unknown authority when trying to download vulnerability DB from private Registry

### ISSUE THREAD

**Issue by mauricewittek** [aquasecurity/trivy-operator#127]
**What steps did you take and what happened:**

I installed the trivy-operator via kubectl in version [0.0.5](https://aquasecurity.github.io/trivy/v0.28.1/docs/kubernetes/operator/installation/kubectl/) and made the necessary changes to pull all images from our internal registry (JFrog Artifactory). The operator is running fine and starts the scanning pods. Except for the changes to the image sources, the operator is running with the default config.

However, the logs show the following error: `failed to download vulnerability DB: OCI artifact error: OCI artifact error: OCI repository error: Get \"https://packages.repo.internal.com/v2/\": x509: certificate signed by unknown authority\n"`

This repository uses a self-signed certificate, so this error is expected.

In the [documentation](https://aquasecurity.github.io/trivy-operator/v0.0.1/vulnerability-scanning/trivy/) I found a parameter to set registries to unsecure: trivy.insecureRegistry.<id>

So I added this to the `trivy-operator-trivy-config` configmap:
```yaml
data:  
  trivy.insecureRegistry.artifactory: "packages.repo.internal.com"
```

This seems to have no effect, the error is unchanged.

**What did you expect to happen:**

I would expect trivy to skip the certificate validation with this setting.

**Anything else you would like to add:**

**Environment:**

- Trivy-Operator version (use `trivy-operator version`): 0.0.5
- Kubernetes version (use `kubectl version`): v1.22.6
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc): Ubuntu 20.04

**Comment by mauricewittek** [comment]
> have you tried adding this param in configmap : trivy.serverInsecure: true ?

From my understanding this setting would not apply in standalone mode, or am I mistaken? I thought it would only apply to client server mode with an external trivy server.

I have now tried to add this setting, however the error is the same. I tried both settings at the same time and both settings separately, but I still get the same error in any case.

**Comment by chen-keinan** [comment]
its look like a proxy or else is redirecting the download URL to `https://packages.repo.internal.com/v2/` , have you tried to use trivy by it self without operator and see if you get the same issue?

**Comment by mauricewittek** [comment]
I changed `trivy.dbRepository` in the deployment manifest from `ghcr.io/aquasecurity/trivy-db` to `packages.repo.internal.com/github-proxy/aquasecurity/trivy-db`, as I'm deploying trivy in an airgapped environment so I cannot access ghcr.io directly.

For the same reason trivy by itself only works if I download the vulnerability database manually and run trivy with `--skip-update --offline-scan`

**Comment by chen-keinan** [comment]
Trivy has added support for it [trivy#240](https://github.com/aquasecurity/trivy/pull/2140), once it will be released we will update trivy version in trivy-operator

**Comment by chen-keinan** [comment]
@mauricewittek FYI #164

**Comment by mauricewittek** [comment]
I just tested with the new Trivy-Operator 0.0.8 which uses a trivy 0.29.1, but I still get the same error:

`{"level":"error","ts":1655803858.6775062,"logger":"reconciler.vulnerabilityreport","msg":"Scan job container","job":"trivy-system/scan-vulnerabilityreport-6f9895d478","container":"39edf7f4-6bff-4611-9e40-fb0e586e23b9","status.reason":"Error","status.message":"2022-06-21T09:30:57.562Z\t\u001b[34mINFO\u001b[0m\tNeed to update DB\n2022-06-21T09:30:57.562Z\t\u001b[34mINFO\u001b[0m\tDB Repository: packages.repo.internal.com/docker-github-proxy/aquasecurity/trivy-db\n2022-06-21T09:30:57.562Z\t\u001b[34mINFO\u001b[0m\tDownloading DB...\n2022-06-21T09:30:57.611Z\t\u001b[31mFATAL\u001b[0m\tinit error: DB error: failed to download vulnerability DB: OCI artifact error: OCI artifact error: OCI repository error: Get \"https://packages.repo.internal.com/v2/\": x509: certificate signed by unknown authority\n","stacktrace":"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport.(*WorkloadController).reconcileJobs.func1\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller.go:323\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile/reconcile.go:102\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:121\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:320\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:234"}
`

Am I missing something in my configuration? I've set the following parameters in the deployment manifest:

```yaml
trivy.insecureRegistry.artifactory: "packages.repo.internal.com"
trivy.serverInsecure: "true"
```

**Comment by chen-keinan** [comment]
@mauricewittek you mention that you are air-gapped env. where do you store the vuln db you download  ?

**Comment by chen-keinan** [comment]
@mauricewittek join our [slack channel]( https://slack.aquasec.com/ ) to discuss in details

**Comment by mauricewittek** [comment]
We are using jFrog Artifactory as a Proxy for external Repositories. So if I want to use the image `docker.io/aquasec/trivy-operator:0.0.8` I have to change the address to `packages.repo.internal.com/docker-proxy/aquasec/trivy-operator:0.0.8`. And this works, the image can be pulled and the operator pod is running in the cluster.

I used the same logic to change `trivy.dbRepository` from 

```yaml
  trivy.dbRepository: "ghcr.io/aquasecurity/trivy-db"
```
to

```yaml
 trivy.dbRepository: "packages.repo.internal.com/docker-github-proxy/aquasecurity/trivy-db"
```

Assuming that the db could be pulled then.

**Comment by chen-keinan** [comment]
when you resolve it with CLI it works I assume ,which repo type you define and what URL you put in ?

**Comment by mauricewittek** [comment]
I'm afraid I don't understand what you mean with 

> when you resolve the vulndb with trivy CLI it works I assume 

But I tried the documentation for using Trivy in Air-Gapped environments: https://aquasecurity.github.io/trivy/v0.29.1/docs/advanced/air-gap/#run-trivy-with-skip-update-and-offline-scan-option

And a `$ oras pull packages.repo.internal.com/docker-github-proxy/aquasecurity/trivy-db:2 -a` works fine and downloads the `db.tar.gz`.

**Comment by nfsouzaj** [comment]
@chen-keinan Hi, I am also having the same issue. In my case the SSL Inspection of Palo Alto messes up with the certificate which causes it to become "insecure". I am very much looking for the insecure flag so that I can set it in the config map and use the operator seamless. Thanks!

**Comment by nfsouzaj** [comment]
@chen-keinan  Can you please reopen the ticket. I am still facing the same issue. 
I downloaded the latest operator, added the flag` trivy.serverInsecure: "true"` and set the trivy version to `trivy.imageRef: "docker.io/aquasec/trivy:0.29.2"` within the configmap trivy-operator-trivy-config and I am still getting the same message: 

`tinit error: DB error: failed to download vulnerability DB: OCI artifact error: OCI artifact error: OCI repository error: Get \"https://ghcr.io/v2/\": x509: certificate signed by unknown authority\n","stacktrace":"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport.(*WorkloadController).reconcileJobs.func1\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller.go:323\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile/reconcile.go:102\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:121\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:320\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:234"}`

**Comment by chen-keinan** [comment]
> @chen-keinan Can you please reopen the ticket. I am still facing the same issue. I downloaded the latest operator, added the flag` trivy.serverInsecure: "true"` and set the trivy version to `trivy.imageRef: "docker.io/aquasec/trivy:0.29.2"` within the configmap trivy-operator-trivy-config and I am still getting the same message:
> 
> `tinit error: DB error: failed to download vulnerability DB: OCI artifact error: OCI artifact error: OCI repository error: Get \"https://ghcr.io/v2/\": x509: certificate signed by unknown authority\n","stacktrace":"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport.(*WorkloadController).reconcileJobs.func1\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller.go:323\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile/reconcile.go:102\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:121\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:320\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.1/pkg/internal/controller/controller.go:234"}`

@nfsouzaj  this issue has not yet been released , it will be released with trivy-operator v0.0.9

**Comment by chen-keinan** [comment]
@nfsouzaj [trivy-operator v0.1.0](https://github.com/aquasecurity/trivy-operator/discussions/229) has been released checkout the fixes in notes
